### PR TITLE
ICDS monthly register new indicators

### PR DESCRIPTION
**Pull request by lwyszomi**
Hi @czue,

I added a new indicator to the monthly register report, and I improved part of the code. 
More details: https://manage.dimagi.com/default.asp?266157

Please let me know if you have any questions.

Regards,
Łukasz

### CODE REVIEW

File: custom/icds_reports/reports/issnip_monthly_register.py [hunk 1]
```diff
@@ -289,32 +368,23 @@ def agg_child_health_monthly(self):
...
-            infrastructure_data = [x for x in self.infrastructure_data if x['awc_id'] == awc]
-            vhnd_data = [x for x in self.vhnd_data if x['awc_id'] == awc]
-            agg_child_health_monthly = [x for x in self.agg_child_health_monthly if x['awc_id'] == awc]
             yield dict(
                 awc_name=self.get_awc_name(awc),
-                agg_awc_monthly_data=agg_awc_monthly_data[0] if agg_awc_monthly_data else None,
-                child_health_monthly_data=child_health_monthly_data[0] if child_health_monthly_data else None,
-                css_record_monthly=css_record_monthly[0] if css_record_monthly else None,
-                infrastructure_data=infrastructure_data[-1] if infrastructure_data else None,
-                vhnd_data=vhnd_data[-1] if vhnd_data else None,
```
Comment (czue) [inline custom/icds_reports/reports/issnip_monthly_register.py]: I noticed that some of these were indexing `-1` while others were indexing `0` but it seems the code expects there to only ever be one of these anyways. was that just a legacy thing that was no longer actually doing anything?

File: custom/icds_reports/models.py [hunk 2]
```diff
@@ -862,3 +864,77 @@ def aggregate(cls, state_id, month):
             cursor.execute(helper.drop_table_query())
             cursor.execute(curr_month_query, curr_month_params)
             cursor.execute(agg_query, agg_params)
+
+
+class ChildHealthMonthly(models.Model):
```
Comment (czue) [inline custom/icds_reports/models.py]: probably revealing my ignorance about how this all works, but where does this table get created? or was it already being created and just had no model?

**Review by czue — APPROVED** [review]
Looks good. Just had a couple questions.

File: custom/icds_reports/models.py (hunk 2 above)
Comment (lwyszomi) [reply custom/icds_reports/models.py]: Hi @czue, sorry for late response. This table is creating in this script https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/migrations/sql_templates/create_tables_and_views.sql#L246

File: custom/icds_reports/reports/issnip_monthly_register.py (hunk 1 above)
Comment (lwyszomi) [reply custom/icds_reports/reports/issnip_monthly_register.py]: @czue, yes when we have ```-1``` we can have a situation where we can get more records but we want only newest

**Comment by czue** [comment]
thanks for responses!

**Pull request merged**